### ISSUE TITLE
Notifications from webpage

### ISSUE THREAD

**Issue by EnderDev** [electron/electron#20312]
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for a feature request that matches the one I want to file, without success.

### Problem Description
I was trying to handle notifications coming from the webpage in the BrowserView, however realised there isn't a way to handle them.

### Proposed Solution
My solution is to have an event which listens for any incoming notification requests.

Example:
```js

BrowserView.webContents.on('before-notification-event', (event, notification) => {
        console.log(notification);
        // Notification {onclick: null, onshow: null, onerror: null, onclose: null, title: "Testing", …}
})

```

### Alternatives Considered
Tried using the preload, but couldn't find a HTML event to listen for notifications.

### Additional Information
N/A

**Comment by MarshallOfSound** [comment]
Notifications are handled entirely in the renderer process, you can easily do `window.Notification = MyCustomNotificationImpl` though if you want to intercept / modify notifications.

**Comment by Shintarot** [comment]
> Notifications are handled entirely in the renderer process, you can easily do `window.Notification = MyCustomNotificationImpl` though if you want to intercept / modify notifications.

I don't think it is a good idea to override `window.Notification`.

First, we need to implement a protocol-compliant `Notification` class; unlike setting up event handlers for `webContents`, this is a lot of work.

Also, I tried overriding `window.Notification` with `preload` but it doesn't work. I ended up using `webContents.executeJavaScript` to override `window.Notification`. This is not very pretty.

Why can't we use event handlers from the `main` process?
